hoot-js/src/main/cpp/hoot/js/conflate/matching/ScriptMatchCreator.cpp

```diff
@@ -398,26 +404,65 @@ class ScriptMatchVisitor : public ConstElementVisitor
     HandleScope handleScope(current);
     Context::Scope context_scope(_script->getContext(current));
     Persistent<Object> plugin(current, getPlugin(_script));
-    Handle<String> isMatchCandidateStr = String::NewFromUtf8(current, "isMatchCandidate");
-    if (ToLocal(&plugin)->Has(isMatchCandidateStr) == false)
-    {
-      throw HootException("Error finding 'isMatchCandidate' function.");
-    }
-    Handle<Value> value = ToLocal(&plugin)->Get(isMatchCandidateStr);
-    if (value->IsFunction() == false)
-    {
-      throw HootException("isMatchCandidate is not a function.");
-    }
-    Handle<Function> func = Handle<Function>::Cast(value);
-    Handle<Value> jsArgs[2];
 
-    int argc = 0;
-    jsArgs[argc++] = getOsmMapJs();
-    jsArgs[argc++] = ElementJs::New(e);
+    bool result = false;
+
+    // Prioritize exports.matchCandidateCriterion over the isMatchCandidate function
```

Comment:
This is the code that fails as described in #3047.  Its ok to leave commented out for now...doesn't affect this change, but if #3047 can be fixed will clean things up in the code quite a bit.